src/utils/readtime.utils.js

```diff
@@ -0,0 +1,72 @@
+import moment from 'moment';
+import momentDurationFormatSetup from 'moment-duration-format';
+import faker from 'faker';
+
+momentDurationFormatSetup(moment);
+
+/**
+ * @description counts the number of words in a string.
+ * excluding spaces, tab and newlines
+ * @param {string} words
+ * @returns {number} number of words
+ */
+export function wordCount(words) {
+  return Math.floor(words.split(/[^\s]+/).length - 1);
+}
+
+/**
+ * @description calculate the time to view images in an article
+ * @param {array} images
+ * @returns {number} time in seconds for all images
+ */
+export function calcImageTime(images) {
+  let initSeconds = 11;
+  let totalSeconds = 0;
+  for (let image of images) {
+    totalSeconds += initSeconds;
+    if (initSeconds > 3) {
+      initSeconds -= 3;
+    }
+  }
+  return totalSeconds;
+}
+
+/**
+ * @description calculate the total average time to read videos
+ * @param {array} videos list the various video durations ['1023', '23023']
+ * @returns {number} minutes to view all videos
+ */
+export function calcVideoTime(videos) {
+  let totalDuration = 0;
+  for (let video of videos) {
+    totalDuration += video;
+  }
+  return totalDuration;
+}
+
+/**
+ * @description calculate the total average time to read only text based article
+ * @param {number} seconds
+ * @returns {number} minutes to read article
+ */
+export function calcTextTime(seconds) {
+  if (!Number(seconds)) return 0;
+  const averageReaderSpeed = 265;
+  const standardMinute = 60;
+  return Math.floor((seconds / averageReaderSpeed) * standardMinute);
+}
+
+/**
+ * @description calculate the total number of time to read a given article
+ * @param {object} content
+ * @returns {string} minutes to read article
+ */
+export function calculateTimeToReadArticle(content) {
+  const imageTime = calcImageTime(content.images);
+  const videoTime = calcVideoTime(content.videos);
+  const numberOfWords = wordCount(content.words);
+  const textTime = calcTextTime(numberOfWords);
+  const seconds = imageTime + videoTime + textTime;
+  const lex = moment.duration(seconds, 'seconds').format('hh:mm:ss');
```

Comment:
What's 'lex' please?

Reply:
that's a typo error. I would remove it. Thanks for that